internal/character/seele/talent.go

```diff
@@ -0,0 +1,66 @@
+package seele
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	Resurgence key.Modifier = "seele-resurgence"
+)
+
+// Enters the buffed state upon defeating an enemy with Basic ATK, Skill, or Ultimate,
+// and receives an extra turn. While in the buffed state,
+// the DMG of Seele's attacks increases by 80% for 1 turn(s).
+// Enemies defeated in the extra turn provided by "Resurgence"
+// will not trigger another "Resurgence."
+
+func init() {
+	modifier.Register(Resurgence, modifier.Config{
+		Stacking:   modifier.ReplaceBySource,
+		StatusType: model.StatusType_STATUS_BUFF,
+	})
+}
+
+func (c *char) initTalent() {
+	// listen to death events. If killer is seele,
+	// enter buffed state and set trigger for extra/resurgence turn.
+	c.engine.Events().TargetDeath.Subscribe(func(e event.TargetDeath) {
+		if e.Killer == c.id {
+			c.enterBuffedState()
+		}
+		c.hasKilled = true
```

Comment:
* Should only set to true if seele is the killer
* This has a problem where if the E6 kills will trigger the buffed state regardless of who's action it is. Not sure what the best solution is at the moment. One idea is you could also have an `ActionStart` subscription that keeps track of the action owner in char and only add buffed state iff `actionStart.Owner == e.Killer == c.id`. Can also add a check of `!e.Insert` here which makes it more game accurate and simplify `ActionEnd` subscription to only look at `hasKilled` boolean (which may want to rename to something more like `shouldInsert`), but imo not strictly necessary.

For 2nd item: I am fine with not fixing it right now and cutting an issue for us to fix it in the future (but obviously best if it just works correctly)